src/layouts/EditHomepage.jsx

```diff
@@ -1171,14 +1173,36 @@ const EditHomepage = ({ match }) => {
   return (
     <>
       {itemPendingForDelete.id && (
-        <DeleteWarningModal
-          onCancel={() => setItemPendingForDelete({ id: null, type: "" })}
-          onDelete={() => {
-            deleteHandler(itemPendingForDelete.id)
+        <GenericWarningModal
+          isOpen={isOpen}
+          onClose={() => {
             setItemPendingForDelete({ id: null, type: "" })
+            onClose()
           }}
-          type={itemPendingForDelete.type}
-        />
+          displayTitle={`Delete ${itemPendingForDelete.type} section`}
+          displayText={`Are you sure you want to delete ${itemPendingForDelete.type}?`}
+        >
+          <Button
+            variant="ghost"
+            colorScheme="secondary"
+            onClick={() => {
+              setItemPendingForDelete({ id: null, type: "" })
+              onClose()
+            }}
+          >
+            Cancel
+          </Button>
+          <Button
+            colorScheme="danger"
+            onClick={() => {
+              deleteHandler(itemPendingForDelete.id)
+              setItemPendingForDelete({ id: null, type: "" })
```

Comment:
hmm this pattern seems to be repeated through out the code - why would we not just inline the 
```ts
setItemPendingForDelete({
  id: event.target.id,
  type,
})
```
here?

Reply:
These actually handle 2 separate tasks - `deleteHandler` actually handles the removal of the content, while `setItemPendingForDelete` indicates the item being selected. Modification of this behaviour is complicated due to the scope of editHomepage - I don't think this should be tackled in this PR